run_test.go

```diff
@@ -184,6 +185,64 @@ func TestRunsCancel(t *testing.T) {
 	})
 }
 
+func TestRunsForceCancel(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	wTest, wTestCleanup := createWorkspace(t, client, nil)
+	defer wTestCleanup()
+
+	// We need to create 2 runs here. The first run will automatically
+	// be planned so that one cannot be cancelled. The second one will
+	// be pending until the first one is confirmed or discarded, so we
+	// can cancel that one.
+	_, _ = createRun(t, client, wTest)
+	rTest, _ := createRun(t, client, wTest)
+
+	t.Run("run is not force-cancelable", func(t *testing.T) {
+		assert.False(t, rTest.Actions.IsForceCancelable)
+	})
+
+	t.Run("user is allowed to force-cancel", func(t *testing.T) {
+		assert.True(t, rTest.Permissions.CanForceCancel)
+	})
+
+	t.Run("after a normal cancel", func(t *testing.T) {
+		// Request the normal cancel
+		err := client.Runs.Cancel(ctx, rTest.ID, RunCancelOptions{})
+		assert.NoError(t, err)
+
+		// Allow time for the async cancel request to operate.
+		time.Sleep(5 * time.Second)
+
+		// Refresh the view of the run
+		rTest, err = client.Runs.Read(ctx, rTest.ID)
+		assert.NoError(t, err)
```

Comment:
The same for this call. Think this also should be `require.NoError`. But if you choose in use a loop to check the actual state before moving on, this call would not be needed here anymore I guess.

Reply:
Good idea. I updated with a loop as suggested below, but I'll keep the error check, because I don't expect errors. The loop is meant to ride out waiting for a certain state to be achieved, not paper over unexpected errors.